### ISSUE TITLE
Switch to h5py for HDF5 output?

### ISSUE THREAD

**Issue by bryanwweber** [bryanwweber/CanSen#42]
Investigate h5py instead of pytables

**Comment by bryanwweber** [comment]
Seems like h5py would be a lot simpler and easier to maintain

**Comment by ischoegl** [comment]
I've used h5py quite a bit, and it is indeed nice and simple. Interestingly, I've considered switching to pytables, mostly to write data files that are compatible with pandas (which does much of the data handling I had hand-coded earlier out of the box). I'd be curious about your thoughts.

**Comment by kyleniemeyer** [comment]
I'm also interested in h5py vs pytables, because https://github.com/pr-omethe-us/PyTeCK also currently uses pytables (following CanSen's lead)

**Comment by bryanwweber** [comment]
I've not looked at this for 2 years... at the time, my thoughts were that the API in h5py is much simpler but the one in PyTables seems to be a lot more powerful. I didn't know there was any difference in terms of Pandas compatible output.

**Comment by ischoegl** [comment]
OK, I figure this is an old post after all. My perspective is that h5py provides maximum flexibility in terms of writing what you want and how you want it. My understanding of pytables (which I haven't used yet and thus triggered this question) is that it uses a subset of functions and enforces a more rigid data structure. Finally, pandas uses pytables as a backend (with some specifics that are largely hidden from the user), but has the price of creating a significant dependency. On the upside, it also supports csv input/output, and extremely convenient data handling. 

I have used h5py for an imaging library we are working on (not public at the moment) and have little reason to change. On the other hand, I found pandas (with the pytables backend) to be extremely convenient for handling of 1D data or time series, and ditched h5py as pandas does all of that out of the box. I used that approach [here](https://github.com/ischoegl/ctwrap).
